hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -439,11 +444,15 @@ class Setting(object):
 
 settings._define_setting(
     'max_shrinks',
-    default=500,
+    default=not_set,
     description="""
 Once this many successful shrinks have been performed, Hypothesis will assume
 something has gone a bit wrong and give up rather than continuing to try to
 shrink the example.
+""",
+    deprecation_message="""
+The max_shrinks setting has been disabled, as internal heuristics are more
+useful for this purpose than a user setting.  It no longer has any effect.
```

Comment:
“It no longer has any effect” – but if it’s set to 0, we don’t set the shrink phase? That doesn’t seem consistent.